teacher.rb

```diff
@@ -2,13 +2,15 @@
 
 class Teacher < Person
   attr_accessor :specialization
+  attr_reader :json_class
 
-  def initialize(specialization, age, name = 'unknown', parent_permission: true)
+  def initialize(age, specialization, name = 'Unknown', parent_permission: true, json_class: 'Teacher')
     super(age, name, parent_permission: parent_permission)
```

Comment:
- Also, in the super, you don't need to specify the parent_permission again, as it's already a value unique to it. you should leave it like this `super(age, name, parent_permission)`. Kindly, improve on this.